From c9d7d2eb73f2c6024e3f94765fc830bce0203f2b Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 1 Apr 2020 17:24:15 +0100
Subject: [PATCH] overlays: Formally rename/deprecate old overlays

Take advantage of the overlay_map to rename or deprecate some obsolete
overlays.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/Makefile           |  7 ----
 arch/arm/boot/dts/overlays/README             | 12 +-----
 .../overlays/bmp085_i2c-sensor-overlay.dts    | 23 -----------
 .../dts/overlays/i2c0-bcm2708-overlay.dts     | 14 -------
 .../dts/overlays/i2c1-bcm2708-overlay.dts     |  9 -----
 arch/arm/boot/dts/overlays/overlay_map.dts    | 40 +++++++++++++++++++
 .../boot/dts/overlays/pi3-act-led-overlay.dts |  1 -
 .../dts/overlays/pi3-disable-bt-overlay.dts   |  1 -
 .../dts/overlays/pi3-disable-wifi-overlay.dts |  1 -
 .../dts/overlays/pi3-miniuart-bt-overlay.dts  |  1 -
 10 files changed, 42 insertions(+), 67 deletions(-)
 delete mode 100644 arch/arm/boot/dts/overlays/bmp085_i2c-sensor-overlay.dts
 delete mode 100644 arch/arm/boot/dts/overlays/i2c0-bcm2708-overlay.dts
 delete mode 100644 arch/arm/boot/dts/overlays/i2c1-bcm2708-overlay.dts
 delete mode 100644 arch/arm/boot/dts/overlays/pi3-act-led-overlay.dts
 delete mode 100644 arch/arm/boot/dts/overlays/pi3-disable-bt-overlay.dts
 delete mode 100644 arch/arm/boot/dts/overlays/pi3-disable-wifi-overlay.dts
 delete mode 100644 arch/arm/boot/dts/overlays/pi3-miniuart-bt-overlay.dts

--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -27,7 +27,6 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	audiosense-pi.dtbo \
 	audremap.dtbo \
 	balena-fin.dtbo \
-	bmp085_i2c-sensor.dtbo \
 	dht11.dtbo \
 	dionaudio-loco.dtbo \
 	dionaudio-loco-v2.dtbo \
@@ -75,9 +74,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	i2c-rtc-gpio.dtbo \
 	i2c-sensor.dtbo \
 	i2c0.dtbo \
-	i2c0-bcm2708.dtbo \
 	i2c1.dtbo \
-	i2c1-bcm2708.dtbo \
 	i2c3.dtbo \
 	i2c4.dtbo \
 	i2c5.dtbo \
@@ -114,10 +111,6 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	mz61581.dtbo \
 	ov5647.dtbo \
 	papirus.dtbo \
-	pi3-act-led.dtbo \
-	pi3-disable-bt.dtbo \
-	pi3-disable-wifi.dtbo \
-	pi3-miniuart-bt.dtbo \
 	pibell.dtbo \
 	piglow.dtbo \
 	piscreen.dtbo \
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -1288,11 +1288,8 @@ Params: pins_0_1                Use pins
 
 
 Name:   i2c0-bcm2708
-Info:   Deprecated, legacy version of i2c0, from which it inherits its
-        parameters, just adding the explicit individual pin specifiers.
+Info:   Deprecated, legacy version of i2c0.
 Load:   <Deprecated>
-Params: sda0_pin                GPIO pin for SDA0 (deprecated - use pins_*)
-        scl0_pin                GPIO pin for SCL0 (deprecated - use pins_*)
 
 
 Name:   i2c1
@@ -1307,13 +1304,8 @@ Params: pins_2_3                Use pins
 
 
 Name:   i2c1-bcm2708
-Info:   Deprecated, legacy version of i2c1, from which it inherits its
-        parameters, just adding the explicit individual pin specifiers.
+Info:   Deprecated, legacy version of i2c1.
 Load:   <Deprecated>
-Params: sda1_pin                GPIO pin for SDA1 (2 or 44 - default 2)
-        scl1_pin                GPIO pin for SCL1 (3 or 45 - default 3)
-        pin_func                Alternative pin function (4 (alt0), 6 (alt2) -
-                                default 4)
 
 
 Name:   i2c3
--- a/arch/arm/boot/dts/overlays/bmp085_i2c-sensor-overlay.dts
+++ /dev/null
@@ -1,23 +0,0 @@
-// Definitions for BMP085/BMP180 digital barometric pressure and temperature sensors from Bosch Sensortec
-/dts-v1/;
-/plugin/;
-
-/ {
-        compatible = "brcm,bcm2835";
-
-        fragment@0 {
-                target = <&i2c_arm>;
-                __overlay__ {
-                        #address-cells = <1>;
-                        #size-cells = <0>;
-                        status = "okay";
-
-                        bmp085@77 {
-                                compatible = "bosch,bmp085";
-                                reg = <0x77>;
-                                default-oversampling = <3>;
-                                status = "okay";
-                        };
-                };
-        };
-};
--- a/arch/arm/boot/dts/overlays/i2c0-bcm2708-overlay.dts
+++ /dev/null
@@ -1,14 +0,0 @@
-#include "i2c0-overlay.dts"
-
-/{
-	__overrides__ {
-		sda0_pin = <&pins1>,"brcm,pins:0",
-			   <&pins2>,"brcm,pins:0",
-			   <&pins3>,"brcm,pins:0",
-			   <&pins4>,"brcm,pins:0";
-		scl0_pin = <&pins1>,"brcm,pins:4",
-			   <&pins2>,"brcm,pins:4",
-			   <&pins3>,"brcm,pins:4",
-			   <&pins4>,"brcm,pins:4";
-	};
-};
--- a/arch/arm/boot/dts/overlays/i2c1-bcm2708-overlay.dts
+++ /dev/null
@@ -1,9 +0,0 @@
-#include "i2c1-overlay.dts"
-
-/{
-	__overrides__ {
-		sda1_pin = <&pins1>,"brcm,pins:0", <&pins2>,"brcm,pins:0";
-		scl1_pin = <&pins1>,"brcm,pins:4", <&pins1>,"brcm,pins:4";
-		pin_func = <&pins1>,"brcm,function:0", <&pins2>,"brcm,function:0";
-	};
-};
--- a/arch/arm/boot/dts/overlays/overlay_map.dts
+++ b/arch/arm/boot/dts/overlays/overlay_map.dts
@@ -1,6 +1,18 @@
 /dts-v1/;
 
 / {
+	bmp085_i2c-sensor {
+		deprecated = "use i2c-sensor,bmp085";
+	};
+
+	i2c0-bcm2708 {
+		deprecated = "use i2c0";
+	};
+
+	i2c1-bcm2708 {
+		deprecated = "use i2c1";
+	};
+
 	i2c3 {
 		bcm2711;
 	};
@@ -17,10 +29,34 @@
 		bcm2711;
 	};
 
+	lirc-rpi {
+		deprecated = "use gpio-ir";
+	};
+
+	pi3-act-led {
+		renamed = "act-led";
+	};
+
+	pi3-disable-bt {
+		renamed = "disable-bt";
+	};
+
+	pi3-disable-wifi {
+		renamed = "disable-wifi";
+	};
+
+	pi3-miniuart-bt {
+		renamed = "miniuart-bt";
+	};
+
 	rpivid-v4l2 {
 		bcm2711;
 	};
 
+	sdio-1bit {
+		deprecated = "use sdio,bus_width=1,gpios_22_25";
+	};
+
 	spi3-1cs {
 		bcm2711;
 	};
@@ -68,4 +104,8 @@
 	uart5 {
 		bcm2711;
 	};
+
+	upstream-aux-interrupt {
+		deprecated = "no longer necessary";
+	};
 };
--- a/arch/arm/boot/dts/overlays/pi3-act-led-overlay.dts
+++ /dev/null
@@ -1 +0,0 @@
-#include "act-led-overlay.dts"
--- a/arch/arm/boot/dts/overlays/pi3-disable-bt-overlay.dts
+++ /dev/null
@@ -1 +0,0 @@
-#include "disable-bt-overlay.dts"
--- a/arch/arm/boot/dts/overlays/pi3-disable-wifi-overlay.dts
+++ /dev/null
@@ -1 +0,0 @@
-#include "disable-wifi-overlay.dts"
--- a/arch/arm/boot/dts/overlays/pi3-miniuart-bt-overlay.dts
+++ /dev/null
@@ -1 +0,0 @@
-#include "miniuart-bt-overlay.dts"
